src/py/mat3ra/made/tools/modify.py

```diff
@@ -1,29 +1,35 @@
 from typing import Union
 
-from ase import Atoms
+from mat3ra.made.material import Material
 from pymatgen.analysis.structure_analyzer import SpacegroupAnalyzer
 from pymatgen.core.structure import Structure
 
-from .convert import (
-    decorator_convert_material_args_kwargs_to_atoms,
-    decorator_convert_material_args_kwargs_to_structure,
-)
+from .convert import decorator_convert_material_args_kwargs_to_structure
 from .utils import translate_to_bottom_pymatgen_structure
 
 
-@decorator_convert_material_args_kwargs_to_atoms
-def filter_by_label(atoms: Atoms, label: Union[int, str]):
+def filter_by_label(material: Material, label: Union[int, str]) -> Material:
     """
-    Filter out only atoms corresponding to the label/tag.
+    Filter out only atoms corresponding to the label.
 
     Args:
-        atoms (ase.Atoms): The Atoms object to filter.
+        material (Material): The material object to filter.
         label (int|str): The tag/label to filter by.
 
     Returns:
-        ase.Atoms: The filtered Atoms object.
+        Material: The filtered material object.
     """
-    return atoms[atoms.get_tags() == label]
+    labels = material.basis["labels"]
+    new_material = material.clone()
+    indices = [idx for idx, _label in enumerate(labels) if _label["value"] == label]
+    new_material.basis["coordinates"] = [
+        coord for idx, coord in enumerate(material.basis["coordinates"]) if idx in indices
+    ]
+    new_material.basis["elements"] = [
+        element for idx, element in enumerate(material.basis["elements"]) if idx in indices
+    ]
+    new_material.basis["labels"] = [label for idx, label in enumerate(labels) if idx in indices]
```

Comment:
Are we repeating this logic here and in https://github.com/Exabyte-io/made/pull/128/files#diff-39a82c4a5b1f8e0d927b57f048c1ac3a4be51c320d77711652f0a07a30845bdbR208? Needs to be isolated